### PR TITLE
A-764 Decco upgrade

### PR DESCRIPTION
**Pull request by shaileshp0110**

### CODE REVIEW

File: src/io/iohk/scalanet/peergroup/SimplestPeerGroup.scala [hunk 1]
```diff
@@ -27,8 +27,8 @@ class SimplestPeerGroup[A, AA, M](
 
   private val log = LoggerFactory.getLogger(getClass)
 
-  private implicit val apc: PartialCodec[A] = aCodec.partialCodec
-  private implicit val aapc: PartialCodec[AA] = aaCodec.partialCodec
+  //private implicit val apc: PartialCodec[A] = aCodec.partialCodec
```
Comment (AlexITC) [inline src/io/iohk/scalanet/peergroup/SimplestPeerGroup.scala]: Delete unused code.

File: WORKSPACE [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+git_repository(
     name = "io_bazel_rules_scala",
-    strip_prefix = "rules_scala-%s" % rules_scala_version,
-    type = "zip",
-    url = "https://github.com/bazelbuild/rules_scala/archive/%s.zip" % rules_scala_version,
+    remote = "https://github.com/bazelbuild/rules_scala",
+    commit = "ca5a7acff4ff630f68f58b8e01e8c25dbf908fb7" # HEAD as of 2019-05-13, update this as needed
+)
+
+http_archive(
```
Comment (AlexITC) [inline WORKSPACE:13]: Protobuf here too.

File: deps [hunk 3]
```diff
@@ -3,7 +3,7 @@
 ROOT=`pwd`
 
 BAZEL_DEPS_REPO="https://github.com/input-output-hk/bazel-deps.git"
-BAZEL_DEPS_COMMIT="b9baec69bdbaae53743a7c0b69828a9c66f6a624"
+BAZEL_DEPS_COMMIT="c9f440d823c987ce6ad86faf10fd53ecf6d9bbc7"
```
Comment (AlexITC) [inline deps:6]: Why do we need the change?

File: WORKSPACE (hunk 2 above)
Comment (shaileshp0110) [reply WORKSPACE:13]: Same as decco and we need to do these changes on multicrypto as well

File: deps (hunk 3 above)
Comment (shaileshp0110) [reply deps:6]: source code scanning this commit is required

**Review by marcesquerra — APPROVED** [review]

**Review by AlexITC — APPROVED** [review]

**Pull request merged**